File: code/accounts/program-derived-address/derived-a-pda/create-program-address.en.ts
import { PublicKey } from "@solana/web3.js";

(async () => {
  const programId = new PublicKey("G1DCNUQTSGHehwdLCAmRyAG8hf51eCHrLNUqkgGKYASj");

  let pda = await PublicKey.createProgramAddress([Buffer.from("test")], programId);
  console.log(pda.toBase58());

  // *failed because the result is on curve
  //     let pda = await PublicKey.createProgramAddress([Buffer.from("config")], programId);
  //     console.log(pda.toBase58());
})();

File: code/accounts/program-derived-address/derived-a-pda/create-program-address.en.rs
use solana_program::pubkey::Pubkey;
use std::str::FromStr;

fn main() {
    let program_id = Pubkey::from_str("G1DCNUQTSGHehwdLCAmRyAG8hf51eCHrLNUqkgGKYASj").unwrap();

    let pda = Pubkey::create_program_address(&[b"test"], &program_id).unwrap();
    println!("pda: {}", pda);
}
